packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -53,17 +148,69 @@ class AnimatedChildSwitcher extends StatefulWidget {
   /// [duration].
   final Widget child;
 
+  /// The duration over which to perform the cross fade using [FadeTransition].
+  final Duration duration;
+
   /// The animation curve to use when fading in the current widget.
   final Curve switchInCurve;
 
   /// The animation curve to use when fading out the previous widgets.
   final Curve switchOutCurve;
 
-  /// The duration over which to perform the cross fade using [FadeTransition].
-  final Duration duration;
+  /// The builder function used to generate custom transitions for
+  /// [AnimatedChildSwitcher].
+  ///
+  /// The default is [AnimatedChildSwitcher.defaultTransitionBuilder].
+  ///
+  /// See also:
+  ///
+  ///  * [AnimatedChildSwitcherTransitionBuilder] for more information about
+  ///    how a transition builder should function.
+  final AnimatedChildSwitcherTransitionBuilder transitionBuilder;
+
+  /// The builder function used to generate custom child layouts for
+  /// [AnimatedChildSwitcher].
+  ///
+  /// The default is [AnimatedChildSwitcher.defaultLayoutBuilder].
+  ///
+  /// See also:
+  ///
+  ///  * [AnimatedChildSwitcherLayoutBuilder] for more information about
+  ///    how a layout builder should function.
+  final AnimatedChildSwitcherLayoutBuilder layoutBuilder;
 
   @override
   _AnimatedChildSwitcherState createState() => new _AnimatedChildSwitcherState();
+
+  /// The default transition algorithm used by [AnimatedChildSwitcher].
+  ///
+  /// The new child is given a [FadeTransition] which increases opacity as
+  /// the animation goes from 0.0 to 1.0, and decreases when the animation is
+  /// reversed.
+  ///
+  /// This is the default value for [transitionBuilder]. It implements
```

Comment:
It doesn't implement anything. If you want to make sure a link to AnimatedChildSwitcherTransitionBuilder ends on in the docs:

The default value for the [transitionBuilder], an [AnimatedChildSwitcherTransitionBuilder] function.

Reply:
Good point.  Done.